### DISCUSSION TITLE
UniGetUI 3.1.0 beta 2

### DISCUSSION THREAD

**Discussion by marticliment** [Devolutions/UniGetUI#2385] [Announcements]
![](https://img.shields.io/github/downloads/marticliment/wingetui/3.1.0-beta2/total?style=for-the-badge&color=red)

<b>IMPORTANT:
This is the third beta release to implement the UniGetUI rebranding.
Instaling this release will automatically migrate settings and other data to new folders (WingetUI -> UniGetUI), so rolling back to older versions of WingetUI may cause settings to be reset.
</b>

NOTE: This release attempts to fix unexpected issues with the dependency installers. Assuming no more issues are found after this release, UniGetUI 3.1.0 stable will be published.
## Changes 
 - Fixes to the dependency installers
 - Improvements to how Microsoft.WinGet.Client is installed by the installer
 - Improvements to the Settings and Log pages.
 - Other bugfixes

SHA256: `81af84438e7203eb6765e43c9bd2d19f915f670852956ca8550de5bac39ad223`

<hr /><em>This discussion was created from the release <a href='https://github.com/marticliment/WingetUI/releases/tag/3.1.0-beta2'>UniGetUI 3.1.0 beta 2</a>.</em>

**Comment by stamminator** [comment]
Is the badge supposed to say `DOWNLOADS@3.1.0-BETA1`?

**Reply by marticliment** [reply]
No

**Comment by schembriaiden** [comment]
When trying to make the Winget package for it again, at times it does the full install then exits and program is recognised as installed by winget instantly. On other times it does not do the full installs and Winget install does not pick up that it has been installed.

### When it is not recognised:

![image](https://github.com/marticliment/WingetUI/assets/81376423/02654b72-9947-4fb2-b529-feea6bc0345f)

### When it is recognised: (I am yet to replicate this again)


The thing holding it back seems to be between installing the dependency for edge webview 2 runtime, and installing the application itself. Its as if 1 install script finishes and it is not continuing to the other install script instantaneously, rather its as if a totally separate install script is running and causes Winget to not recognise it as a continuation of the install.

**Reply by marticliment** [reply]
I am aware of this issue, and I am still investigating why it could be not working

**Comment by JasonUK67** [comment]
Tried to update from v3.0.2 but both this Beta2 & the final release hits the following issue:

On installation/update following message displayed:

Missing dependency. UniGetUI requires WinGet PowerShell Module to operate but it was not found on your system. Please wait while WinGet Powershell Module is being installed.  A black window may show up. Please wait until it closes.


A powershell window opens and quickly displays following:

NuGet provider is required to continue
PowerShellGet requires NuGet provider version '2.8.5.201' or newer to interact with NuGet-based repositories. The NuGet
 provider must be available in 'C:\Program Files\PackageManagement\ProviderAssemblies' or
'C:\Users\Jason\AppData\Local\PackageManagement\ProviderAssemblies'. You can also install the NuGet provider by running
 'Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force'. Do you want PowerShellGet to install and
import the NuGet provider now?
[Y] Yes  [N] No  [S] Suspend  [?] Help (default is "Y"):


UniGet then sits doing nothing, presumably waiting for user response but if you do by accepting default Y to above you get following error messages:

WARNING: Unable to download from URI 'https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x409' to ''.
WARNING: Unable to download the list of available providers. Check your internet connection.
PackageManagement\Install-PackageProvider : No match was found for the specified search criteria for the provider
'NuGet'. The package provider requires 'PackageManagement' and 'Provider' tags. Please check if the specified package
has the tags.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:7468 char:21
+ ...     $null = PackageManagement\Install-PackageProvider -Name $script:N ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (Microsoft.Power...PackageProvider:InstallPackageProvider) [Install-Pac
   kageProvider], Exception
    + FullyQualifiedErrorId : NoMatchFoundForProvider,Microsoft.PowerShell.PackageManagement.Cmdlets.InstallPackagePro
   vider

PackageManagement\Import-PackageProvider : No match was found for the specified search criteria and provider name
'NuGet'. Try 'Get-PackageProvider -ListAvailable' to see if the provider exists on the system.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:7474 char:21
+ ...     $null = PackageManagement\Import-PackageProvider -Name $script:Nu ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (NuGet:String) [Import-PackageProvider], Exception
    + FullyQualifiedErrorId : NoMatchFoundForCriteria,Microsoft.PowerShell.PackageManagement.Cmdlets.ImportPackageProv
   ider

WARNING: Unable to download from URI 'https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x409' to ''.
WARNING: Unable to download the list of available providers. Check your internet connection.
PackageManagement\Get-PackageProvider : Unable to find package provider 'NuGet'. It may not be imported yet. Try
'Get-PackageProvider -ListAvailable'.
At C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:7478 char:30
+ ... tProvider = PackageManagement\Get-PackageProvider -Name $script:NuGet ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Microsoft.Power...PackageProvider:GetPackageProvider) [Get-PackageProvi
   der], Exception
    + FullyQualifiedErrorId : UnknownProviderFromActivatedList,Microsoft.PowerShell.PackageManagement.Cmdlets.GetPacka
   geProvider

Install-Module : NuGet provider is required to interact with NuGet-based repositories. Please ensure that '2.8.5.201'
or newer version of NuGet provider is installed.
At line:1 char:4
+ & {Install-Module -Name Microsoft.WinGet.Client -Force -Confirm:$fals ...
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Install-Module], InvalidOperationException
    + FullyQualifiedErrorId : CouldNotInstallNuGetProvider,Install-Module
Press Enter to continue


Not sure where to go from here as can't proceed beyond this point.

**Reply by marticliment** [reply]
Do you have any AV/Firewall that could be blocking microsoft.com?

**Reply by JasonUK67** [reply]
I deactivated all AV / Windows hardening tools and tried again with same result.  Firewall is standard firewall provided with Windows (I'm still on Windows 10).

**Reply by marticliment** [reply]
Can you access `https://go.microsoft.com/fwlink/?LinkID=627338&clcid=0x409` from your browser?

**Reply by JasonUK67** [reply]
SOLVED:  Disabled Firewall temporarily and UniGet was able to do what was necessary.  Thanks for quick reply/pointing me in right direction

**Comment by Biyakuga** [comment]
Is there a technical reason why updates are forced on users? Unexpectedly, WingetUi updated while I was in the middle of something important. I would like to understand why this behavior is enforced in your software.

**Reply by marticliment** [reply]
It is not "forced", you can disable it from the Settings.
However, it is enabled by default since sometimes I have to push upgrades after some changes on a package manager that break UniGetUI, and if updates are not enforced users don't update and I receive complaints about issues that have already been fixed.

**Reply by Biyakuga** [reply]
I understand that, but why not give us, the users, the ability to choose when to install it, like almost any other software or app?
For example, when there's an update for, say, PowerToys, I receive a notification, and I can download and install it at a convenient time for me. Don't you agree that's a more user-friendly approach?

**Reply by marticliment** [reply]
You can enable this behaviour by disabling automatic UniGetUI updates. However, what you are arguing makes sense, I will consider it and see how it can be implemented

**Reply by Biyakuga** [reply]
Thank you, I really appreciate it.
By the way I edited my comment with an example, I hope you saw it.
